static/js/containers/pages/register/RegisterExtraDetailsPage.js

```diff
@@ -45,16 +49,17 @@ class RegisterExtraDetailsPage extends React.Component<Props> {
       params: { partialToken }
     } = this.props
 
+    /* eslint-disable camelcase */
     try {
       const {
-        body: { state, errors }
-      }: { body: AuthResponse } = await registerExtraDetails(
+        body: { state, errors, redirect_url }
+      }: { body: AuthResponseRaw } = await registerExtraDetails(
         profileData,
         partialToken
       )
 
       if (state === STATE_SUCCESS) {
-        window.location.href = routes.root
+        window.location.href = redirect_url || routes.dashboard
```

Comment:
`redirect_url` seems to be set to `/` if I go to the signup page without the `next` parameter set (for instance from the 'create account' link on the catalog page.  Should `next` be assigned for all links to the create-account url?

Reply:
Hmmm, let me look at that